LockExample.UnitTests/FileCopyTest.cs

```diff
@@ -0,0 +1,44 @@
+namespace LockExample.UnitTests
+{
+    public class FileCopyTest
+    {
+        private const string TempFolderName = "LockTempFilesFolder";
+        private const string DestinationTempFolderName = "Destination";
+
+        [Fact]
+        public async void Copied_Files_To_Destination_Not_Null()
+        {           
+            var tempFolder = Path.GetTempPath();            
+            var path = Path.Combine(tempFolder, TempFolderName);
+            var destPath = Path.Combine(tempFolder, DestinationTempFolderName);
+            if (!Directory.Exists(path))
+            {
+                Directory.CreateDirectory(path);
+                WriteTestFiles(path);
+            }
+            if (!Directory.Exists(destPath))
+            {
+                Directory.CreateDirectory(destPath);
+            }
+            var fileService = new FileService(tempFolder);
+            await fileService.StartFileCopy(tempFolder, DestinationTempFolderName, TempFolderName);
+            var files = Directory.GetFiles(destPath);
+
+            Assert.NotNull(files);
```

Comment:
the test should check if the files are created and content of result file.

The test files should be better removed after the test is run.